samples/core/Miscellaneous/NewInEFCore7/DocumentsContext.cs

```diff
@@ -0,0 +1,595 @@
+        IssueNumber = issueNumber;
+    }
+
+    public int IssueNumber { get; set; }
+    public decimal? CoverPrice { get; set; }
+    public Person Editor { get; set; } = null!;
+}
+
+public class Person
+{
+    public Person(string name)
+    {
+        Name = name;
+    }
+
+    public int Id { get; private set; }
+
+    [ConcurrencyCheck]
+    public string Name { get; set; }
+
+    public ContactDetails Contact { get; set; } = null!;
+
+    public List<Book> PublishedWorks { get; } = new();
+    public List<Magazine> Edited { get; } = new();
+}
+
+public abstract class DocumentsContext : DbContext
+{
+    public bool LoggingEnabled { get; set; }
+    public abstract MappingStrategy MappingStrategy { get; }
+    public virtual bool UseStoredProcedures => true;
+
+    public DbSet<Document> Documents => Set<Document>();
+    public DbSet<Book> Books => Set<Book>();
+    public DbSet<Magazine> Magazines => Set<Magazine>();
+    public DbSet<Person> People => Set<Person>();
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder.UseSqlServer(@$"Server=(localdb)\mssqllocaldb;Database={GetType().Name}")
+            .EnableSensitiveDataLogging()
+            .LogTo(
+                s =>
+                {
+                    if (LoggingEnabled)
+                    {
+                        Console.WriteLine(s);
+                    }
+                }, LogLevel.Information);
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Document>(
+            entityTypeBuilder =>
+            {
+                entityTypeBuilder.Property(document => document.FirstRecordedOn).HasDefaultValueSql("getutcdate()");
+                entityTypeBuilder.Property(document => document.RetrievedOn).HasComputedColumnSql("getutcdate()");
+
+                if (UseStoredProcedures)
+                {
+                    switch (MappingStrategy)
+                    {
+                        case MappingStrategy.Tph:
+                            entityTypeBuilder.InsertUsingStoredProcedure(
+                                "Document_Insert",
+                                storedProcedureBuilder =>
+                                {
+                                    storedProcedureBuilder.HasParameter("Discriminator");
+                                    storedProcedureBuilder.HasParameter(document => document.Title);
+                                    storedProcedureBuilder.HasParameter(document => document.NumberOfPages);
+                                    storedProcedureBuilder.HasParameter(document => document.PublicationDate);
+                                    storedProcedureBuilder.HasParameter(document => document.CoverArt);
+                                    storedProcedureBuilder.HasResultColumn(document => document.Id);
+                                    storedProcedureBuilder.HasParameter(document => ((Book)document).Isbn);
```

Comment:
You can also specify the parameter type
```suggestion
                                    storedProcedureBuilder.HasParameter((Book document) => document.Isbn);
```